src/simulators/statevector/qubitvector.hpp

```diff
@@ -37,10 +37,13 @@ using uint_t = uint64_t;
 using int_t = int64_t;
 using reg_t = std::vector<uint_t>;
 using indexes_t = std::unique_ptr<uint_t[]>;
-using complex_t = std::complex<double>;
-using cvector_t = std::vector<complex_t>;
-using rvector_t = std::vector<double>;
 template <size_t N> using areg_t = std::array<uint_t, N>;
+template <typename T> using complex_t = std::complex<T>;
+template <typename T> using cvector_t = std::vector<complex_t<T>>;
+template <typename T> using rvector_t = std::vector<T>;
+using complexd_t = std::complex<double>;
```

Comment:
Rather than having so many aliases can we use default template arguments like: 

```c++
template <typename T = double> using complex_t = std::complex<T>;
template <typename T = double> using cvector_t = std::vector<complex_t<T>>;
template <typename T = double> using rvector_t = std::vector<T>;
```
Then remove the `complexd_t/cvectord_t/rvectord_t` etc and replace with `cvector_t<>`